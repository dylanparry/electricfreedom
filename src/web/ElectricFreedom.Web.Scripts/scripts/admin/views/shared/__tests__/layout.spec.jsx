import React from 'react';
import { shallow } from 'enzyme';

import Layout from '../layout';

const setup = () =>
{
  const wrapper = shallow(<Layout />);

  return {
    wrapper,
  };
};

describe('Layout', () =>
{
  test('render', () =>
  {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
