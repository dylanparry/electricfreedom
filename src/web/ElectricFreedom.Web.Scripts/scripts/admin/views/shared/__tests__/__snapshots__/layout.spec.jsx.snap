// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <p>
        My App
    </p>
    <menu />
    <Switch>
        <Route
            component={[Function]}
            path="/users"
        />
        <Route
            component={[Function]}
            path="/comments"
        />
        <Route
            component={[Function]}
            path="/reviews"
        />
        <Route
            component={[Function]}
            path="/tags"
        />
        <Route
            component={[Function]}
            path="/articles"
        />
        <Route
            component={[Function]}
            path="/artists/:artistId/albums"
        />
        <Route
            component={[Function]}
            path="/artists"
        />
        <Route
            component={[Function]}
            path="/"
        />
    </Switch>
</div>,
  "nodes": Array [
    <div>
      <p>
            My App
      </p>
      <menu />
      <Switch>
            <Route
                  component={[Function]}
                  path="/users"
            />
            <Route
                  component={[Function]}
                  path="/comments"
            />
            <Route
                  component={[Function]}
                  path="/reviews"
            />
            <Route
                  component={[Function]}
                  path="/tags"
            />
            <Route
                  component={[Function]}
                  path="/articles"
            />
            <Route
                  component={[Function]}
                  path="/artists/:artistId/albums"
            />
            <Route
                  component={[Function]}
                  path="/artists"
            />
            <Route
                  component={[Function]}
                  path="/"
            />
      </Switch>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <layout />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <p>
                    My App
          </p>
          <menu />
          <Switch>
                    <Route
                              component={[Function]}
                              path="/users"
                    />
                    <Route
                              component={[Function]}
                              path="/comments"
                    />
                    <Route
                              component={[Function]}
                              path="/reviews"
                    />
                    <Route
                              component={[Function]}
                              path="/tags"
                    />
                    <Route
                              component={[Function]}
                              path="/articles"
                    />
                    <Route
                              component={[Function]}
                              path="/artists/:artistId/albums"
                    />
                    <Route
                              component={[Function]}
                              path="/artists"
                    />
                    <Route
                              component={[Function]}
                              path="/"
                    />
          </Switch>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <p>
                    My App
          </p>
          <menu />
          <Switch>
                    <Route
                              component={[Function]}
                              path="/users"
                    />
                    <Route
                              component={[Function]}
                              path="/comments"
                    />
                    <Route
                              component={[Function]}
                              path="/reviews"
                    />
                    <Route
                              component={[Function]}
                              path="/tags"
                    />
                    <Route
                              component={[Function]}
                              path="/articles"
                    />
                    <Route
                              component={[Function]}
                              path="/artists/:artistId/albums"
                    />
                    <Route
                              component={[Function]}
                              path="/artists"
                    />
                    <Route
                              component={[Function]}
                              path="/"
                    />
          </Switch>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <layout />,
}
`;
